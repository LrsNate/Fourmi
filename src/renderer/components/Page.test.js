import { mount, render } from "enzyme";
import React from "react";

import {
  mapDispatchToProps,
  mapStateToProps,
  Page
} from "./Page";

jest.mock("react-router-redux", () => ({
  goBack() {
    return { type: "GO_BACK" };
  }
}));

const props = {
  classes: {
    appBarLeftButton: "appBarLeftButton",
    pageContent: "pageContent"
  },
  title: "Some page"
};

describe("The Page component", () => {
  describe("on an initial load", () => {
    it("matches the snapshot", () => {
      expect(
        render(
          <Page {...props}>
            <p>Some content</p>
          </Page>
        )
      ).toMatchSnapshot();
    });
  });

  describe("with the default goBack handler", () => {
    it("does nothing when clicking the back button", () => {
      const wrapper = mount(<Page {...props} />);
      wrapper.find("button.appBarLeftButton").simulate("click");
    });
  });

  describe("with a provided goBack handler", () => {
    it("calls the handler when clicking the back button", () => {
      const goBack = jest.fn(() => {});
      const wrapper = mount(<Page {...props} goBack={goBack} />);
      wrapper.find("button.appBarLeftButton").simulate("click");

      expect(goBack).toHaveBeenCalledTimes(1);
    });
  });
});

describe("The mapStateToProps function", () => {
  it("returns the original props", () => {
    expect(mapStateToProps({ a: "a" }, { b: "b" })).toEqual({ b: "b" });
  });
});

describe("The mapDispatchToProps function", () => {
  it("dispatches the goBack function", () => {
    const dispatch = jest.fn(i => i);
    const { goBack } = mapDispatchToProps(dispatch);

    goBack();

    expect(dispatch).toHaveBeenCalledWith({ type: "GO_BACK" });
  });
});
