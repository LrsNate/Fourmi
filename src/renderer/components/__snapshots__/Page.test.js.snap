// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Page component on an initial load matches the snapshot 1`] = `
<div>
  <header
    class="MuiPaper-root-9 MuiPaper-shadow4-15 MuiAppBar-root-1 MuiAppBar-positionFixed-2 MuiAppBar-colorPrimary-7 mui-fixed"
  >
    <div
      class="MuiToolbar-root-36 MuiToolbar-gutters-37"
    >
      <button
        class="MuiButtonBase-root-44 MuiIconButton-root-38 MuiIconButton-colorInherit-39 appBarLeftButton"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label-43"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-46 MuiSvgIcon-fontSize-52"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root-53"
        />
      </button>
      <h2
        class="MuiTypography-root-60 MuiTypography-title-66 MuiTypography-colorInherit-79"
      >
        Some page
      </h2>
      <button
        class="MuiButtonBase-root-44 MuiIconButton-root-38 MuiIconButton-colorInherit-39"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label-43"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-46 MuiSvgIcon-fontSize-52"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19.43 12.98c.04-.32.07-.64.07-.98s-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46c-.12-.22-.39-.3-.61-.22l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65C14.46 2.18 14.25 2 14 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1c-.23-.09-.49 0-.61.22l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98s.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46c.12.22.39.3.61.22l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.23.09.49 0 .61-.22l2-3.46c.12-.22.07-.49-.12-.64l-2.11-1.65zM12 15.5c-1.93 0-3.5-1.57-3.5-3.5s1.57-3.5 3.5-3.5 3.5 1.57 3.5 3.5-1.57 3.5-3.5 3.5z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root-53"
        />
      </button>
    </div>
  </header>
  <div
    class="pageContent"
  >
    <p>
      Some content
    </p>
  </div>
</div>
`;
